src/libraries/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -107,6 +107,98 @@ public void CheckEndReachableDoesNotCrossPastEnd()
 
         #endregion
 
+        #region Offset
+
+        [Fact]
+        public void GetOffset_SingleSegment()
+        {
+            var buffer = new ReadOnlySequence<T>(new T[50]);
+            Assert.Equal(25, buffer.GetOffset(buffer.GetPosition(25)));
+        }
+
+        private (BufferSegment<T> bufferSegment1, BufferSegment<T> bufferSegment4) GetBufferSegment()
+        {
+            var bufferSegment1 = new BufferSegment<T>(new T[50]);
+            BufferSegment<T> bufferSegment2 = bufferSegment1.Append(new T[50]);
+            BufferSegment<T> bufferSegment3 = bufferSegment2.Append(new T[0]);
+            return (bufferSegment1, bufferSegment3.Append(new T[50]));
+        }
+
+        private ReadOnlySequence<T> GetFourSegmentsReadOnlySequence()
+        {
+            (BufferSegment<T> bufferSegment1, BufferSegment<T> bufferSegment4) = GetBufferSegment();
+            return new ReadOnlySequence<T>(bufferSegment1, 0, bufferSegment4, 50);
+        }
+
+        [Fact]
+        public void GetOffset_MultiSegment_FirstSegment()
+        {
+            ReadOnlySequence<T> buffer = GetFourSegmentsReadOnlySequence();
+            Assert.Equal(25, buffer.GetOffset(buffer.GetPosition(25)));
+        }
+
+        [Fact]
+        public void GetOffset_MultiSegment_LastSegment()
+        {
+            ReadOnlySequence<T> buffer = GetFourSegmentsReadOnlySequence();
+            Assert.Equal(125, buffer.GetOffset(buffer.GetPosition(125)));
+        }
+
+        [Fact]
+        public void GetOffset_MultiSegment_MiddleSegment()
+        {
+            ReadOnlySequence<T> buffer = GetFourSegmentsReadOnlySequence();
+            Assert.Equal(75, buffer.GetOffset(buffer.GetPosition(75)));
+        }
+
+        [Fact]
+        public void GetOffset_SingleSegment_NullPositionObject()
+        {
+            var buffer = new ReadOnlySequence<T>(new T[50]);
+            Assert.Equal(0, buffer.GetOffset(new SequencePosition(null, 25)));
```

Comment:
Should we throw in this case since the `SequencePosition` doesn't belong to this `ReadOnlySequence` and also that it is `null`? To me, this looks similar to the `GetOffset_MultiSegment_InvalidSequencePositionSegment` case where we throw `InvalidCastException`.

**Edit:** Nevermind, other APIs treat `null` object within the `SequencePosition` the same as `_startObject`.